hazelcast/src/main/java/com/hazelcast/nio/Connection.java

```diff
@@ -127,7 +126,36 @@
      * Pending packets on this connection are discarded
      * <p>
      * If the Connection is already closed, the call is ignored. So it can safely be called multiple times.
+     *
+     * @param reason the reason this connection is going to be closed. Is allowed to be null.
+     * @param cause the Throwable responsible for closing this connection. Is allowed to be null.
      */
-    void close();
```

Comment:
you should adjust [this](https://github.com/jerrinot/hazelcast/blob/774ce2c9e17c1ae339c34a685bd814d07d5fd0b1/hazelcast/src/main/java/com/hazelcast/nio/tcp/SocketWriter.java#L85) too